src/utils/Fetcher.ts

```diff
@@ -31,6 +32,51 @@ export class Fetcher {
     )
   }
 
+  async fetchPipe(
+    url: string,
+    writeTo: ReadableStream<Uint8Array>,
+    options?: RequestOptions
+  ): Promise<Map<string, string>> {
+    return this.fetchInternal(
+      url,
+      (response) => this.copyResponse(response, writeTo),
+      this.completeOptionsWithDefault(FETCH_BUFFER_DEFAULTS, options)
+    )
+  }
+
+  private KNOWN_HEADERS: string[] = [
+    'Content-Type',
+    'Access-Control-Allow-Origin',
+    'Access-Control-Expose-Headers',
+    'ETag',
+    'Date',
+    'Content-Length',
+    'Cache-Control'
+  ]
+
+  private async copyResponse(response: Response, writeTo: ReadableStream<Uint8Array>): Promise<Map<string, string>> {
+    // The method pipeTo() is not working, so we need to use pipe() which is the one implemented
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    response.body.pipe(writeTo)
+    return this.onlyKnownHeaders(response)
+  }
+
+  private fixHeaderNameCase(headerName: string): string | undefined {
```

Comment:
Let's try to move this header copying/sanitising into the `catalyst-client`, since it's catalyst-client specific. In here, we can just return all headers